contracts/p1/mixins/Trading.sol

```diff
@@ -120,8 +121,12 @@ abstract contract TradingP1 is Multicall, ComponentP1, ReentrancyGuardUpgradeabl
         IERC20 sell = req.sell.erc20();
         assert(address(trades[sell]) == address(0));
 
-        IERC20Upgradeable(address(sell)).safeApprove(address(broker), 0);
-        IERC20Upgradeable(address(sell)).safeApprove(address(broker), req.sellAmount);
+        // Set allowance via custom approval -- first sets allowance to 0, then sets allowance
+        // to either the requested amount or the maximum possible amount, if that fails.
+        //
+        // Context: wcUSDCv3 has a non-standard approve() function that reverts if the approve
+        // amount is > 0 and < type(uint256).max.
+        AllowanceLib.safeApproveFallbackToMax(req.sell.erc20(), address(broker), req.sellAmount);
```

Comment:
If we are specifically making the changes for Trading stuff, should we consider doing the same in other places?

Reply:
We should have hit all the places that we do an approval for a non-type(uint256).max amount.

Reply:
the only other place would be RevenueTrader._distributeTokenToBuy(), which should only be dealing with an RToken or RSR.

maybe its worth adding a comment about assuming `tokenToBuy` is RToken or RSR?